runtime/src/main/cpp/Memory.h

```diff
@@ -455,6 +466,8 @@ extern "C" {
     ObjHeader* result = name(__VA_ARGS__, OBJ_RESULT);  \
     return result;                                      \
   }
+#define HEAP_RETURN_SLOT(slot) \
```

Comment:
Why not enforce return slot to be always stack-located instead?
I believe that this is
1) possible
2) simpler
3) more efficient

Reply:
It's not just stack allocated, it must be in slots seen when walking the stack. Let's first stabilize this design and see its performance implications and the proceed with optimisations.

Reply:
Did that.

Reply:
Did that help?